arangod/Aql/RemoteExecutor.cpp

```diff
@@ -454,18 +454,15 @@ Result handleErrorResponse(network::EndpointSpec const& spec, fuerte::Error err,
                            fuerte::Response* response) {
   TRI_ASSERT(err != fuerte::Error::NoError || response->statusCode() >= 400);
 
+  constexpr std::string_view prefix("Error message received from ");
   std::string msg;
-  if (spec.shardId.empty()) {
-    msg.append("Error message received from cluster node '")
-        .append(spec.serverId)
-        .append("': ");
-  } else {
-    msg.append("Error message received from shard '")
-        .append(spec.shardId)
-        .append("' on cluster node '")
-        .append(spec.serverId)
-        .append("': ");
+  // avoid a few reallocations for building the error message string
+  msg.reserve(128);
+  msg.append(prefix);
+  if (!spec.shardId.empty()) {
+    msg.append("shard '").append(spec.shardId).append("' from ");
```

Comment:
To generate identical error message as before:
```suggestion
    msg.append("shard '").append(spec.shardId).append("' on ");
```